pandas/io/formats/format.py

```diff
@@ -1667,7 +1667,7 @@ def _format_strings(self) -> list[str]:
 
 
 def format_percentiles(
-    percentiles: (np.ndarray | list[int | float] | list[float] | list[str | float]),
+    percentiles: (np.ndarray | Sequence[int | float]),
```

Comment:
I think you can use just `Sequence[float]` (I think type checkers wiredly treat int as a sub-class of float). If you need the union, please use `Sequence[float] | Sequence[int]`.

Reply:
That seems to be correct: https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html
I adapted the code.